onebusaway-android/src/main/java/org/onebusaway/android/util/DBUtil.java

```diff
@@ -25,4 +29,47 @@ public static void addToDB(ObaStop stop) {
         }
         ObaContract.Stops.insertOrUpdate(stop.getId(), values, true);
     }
+
+    public static void addRouteToDB(Context ctx, ArrivalInfo arrivalInfo){
+        ContentValues routeValues = new ContentValues();
+
+        String shortName = arrivalInfo.getInfo().getShortName();
+        String longName = arrivalInfo.getInfo().getRouteLongName();
+
+        if (TextUtils.isEmpty(longName)) {
+            longName = UIUtils.formatDisplayText(arrivalInfo.getInfo().getHeadsign());
+        }
+
+        routeValues.put(ObaContract.Routes.SHORTNAME, shortName);
+        routeValues.put(ObaContract.Routes.LONGNAME, longName);
+
+        if (Application.get().getCurrentRegion() != null) {
```

Comment:
It seems to me that this check should be moved up to the very top of the method as a precondition for everything else. If the current region is `null`, then just return since there's no reason to save a route to the DB; it'll be unusable.

Reply:
Thanks for correcting me this makes sense, I have submitted a new commit.